youtube_dl/extractor/jamendo.py

```diff
@@ -17,49 +21,44 @@ class JamendoIE(InfoExtractor):
             'display_id': 'stories-from-emona-i',
             'ext': 'flac',
             'title': 'Stories from Emona I',
-            'thumbnail': 're:^https?://.*\.jpg',
-            'url': 'https://flac.jamendo.com/?trackid=196219&format=flac'
+            'thumbnail': 're:^https?://.*\.jpg'
         }
     }
 
     def _real_extract(self, url):
         url_data = self._VALID_URL_RE.match(url)
         track_id = url_data.group('id')
-        webpage = self._download_webpage(url, track_id)
+        display_id = url_data.group('display_id')
+        webpage = self._download_webpage(url, display_id)
 
         thumbnail = self._html_search_meta(
             'image', webpage, 'thumbnail', fatal=False)
         title = self._html_search_meta('name', webpage, 'title')
+
+        url_template = 'https://%s.jamendo.com/?trackid=%s&format=%s&from=app-97dab294'
+        format_data = [
+            FormatData(
+                format_id='mp31', sub_domain='mp3l', ext='mp3', quality=0),
+            FormatData(
+                format_id='mp32', sub_domain='mp3d', ext='mp3', quality=1),
+            FormatData(
+                format_id='ogg1', sub_domain='ogg', ext='ogg', quality=2),
+            FormatData(
+                format_id='flac', sub_domain='flac', ext='flac', quality=3),
```

Comment:
I don't understand your concern.
Do you want me to use this just to not use the word `FormatData` four times?
```python
[
    FormatData(format_id, sub_domain, ext, quality)
    for format_id, sub_domain, ext, quality in [
        ('mp31', 'mp3l', 'mp3', 0),
        ('mp32', 'mp3d', 'mp3', 1),
        ('ogg1', 'ogg', 'ogg', 2),
        ('flac', 'flac', 'flac', 3),
    ]
]
```